WordPress/wordpress-develop
# Legacy widget rendering endpoint

Related to https://github.com/WordPress/gutenberg/issues/33540
Trac Ticket: https://core.trac.wordpress.org/ticket/53957

This adds a `/wp/v2/widget-types/<widget type>/render` endpoint which accepts POST data and returns the same output as the previous iframe-oriented GET endpoint.

Test plan:

1. Go to `/wp-admin/widgets.php`
2. Paste the following script into your devtools:

```js
wp.apiFetch({
      "path": "/index.php?rest_route=%2Fwp%2Fv2%2Fwidget-types%2Ftext%2Frender&context=edit&per_page=100&_locale=user",
      "method": "POST",
      "data": {"id_base":"text","instance":{"encoded":"YTo0OntzOjU6InRpdGxlIjtzOjU6InRlc3QyIjtzOjQ6InRleHQiO3M6NToidGVzdDMiO3M6NjoiZmlsdGVyIjtiOjE7czo2OiJ2aXN1YWwiO2I6MTt9","hash":"6bd53d501ad76cfbc50ef417d78dc9f1","raw":{"title":"test2","text":"test3","filter":true,"visual":true}}}
})
.then(console.log.bind(console))
.catch(console.error.bind(console));
```

3. Confirm the response was successful and the widget got rendered as expected.


**Alternatives considered:**
* `ServerSideRender` – we need to render a full HTML document with `wp_head()`, CSS links, maybe some additional scripts, and this is much more that I would like to return from a block rendering API.
* Updating the existing `admin_init` hook handler to work with `POST` requests – it's tricky and doesn't seem right, I don't think we should stretch hooks like that. This clearly is an API request so let's treat it as such.
